conn.go

```diff
@@ -995,9 +995,11 @@ func (c *Conn) ReadPartitions(topics ...string) (partitions []Partition, err err
 			}
 
 			makeBrokers := func(ids ...int32) []Broker {
-				b := make([]Broker, len(ids))
```

Comment:
I think the original code here was trying to avoid excess allocations. You can do this with your code too, by changing this:

```
var b []Broker
```

to this:

```
b := make([]Broker, 0, len(ids))
```

That form sets the capacity of the slice to `len(ids)` but the length to 0, so your appends still work as expected, but you're guaranteed to avoid additional allocations, at the cost of possibly having a slice which has more capacity than its length.